### ISSUE TITLE
False positive assignment-from-no-return on generator

### ISSUE THREAD

**Issue by xbe** [PyCQA/pylint#3904]
### Steps to reproduce
```
from typing import Callable, Generator

CHUNK_SIZE = 5


def chunk(handler: Callable[[str], None]) -> Generator[None, str, None]:
    buf = ''
    while True:
        while len(buf) < CHUNK_SIZE:
            buf += yield
        head, buf = buf[:CHUNK_SIZE], buf[CHUNK_SIZE:]

        handler(head)


def handle(contents: str) -> None:
    print(f'Got: {contents}')


streamer = chunk(handle)
next(streamer)  # start streaming
streamer.send('foo')
```

### Current behavior
```
************* Module iter
iter.py:20:0: E1111: Assigning result of a function call, where the function has no return (assignment-from-no-return)
```

### Expected behavior
No error.

### pylint --version output
With `pip install pylint astroid --pre -U`:
```
pylint 2.6.0
astroid 2.4.2
Python 3.8.5 (default, Jul 20 2020, 18:32:44) 
[GCC 9.3.0]
```

**Comment by hippo91** [comment]
@xbe thanks for the report. I can reproduce it.

**Comment by PCManticore** [comment]
This will be fixed by https://github.com/PyCQA/astroid/pull/877/files.
